### DISCUSSION TITLE
Custom remote for FIOS

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#117] [General]
I created a FIOS remote, but don't know how to upload it.

**Comment by mdehrlich** [comment]
[FIOS_remote.txt](https://github.com/GerryDazoo/Slinger/files/9729262/FIOS_remote.txt)

**Reply by GerryDazoo** [reply]
Thanks, I've committed it to the main line.

**Comment by mdehrlich** [comment]
Fixed a few problems and cleaned up the positioning on some buttons.
[FIOS_remote.txt](https://github.com/GerryDazoo/Slinger/files/9802256/FIOS_remote.txt)

**Comment by GR0DZ1LLA** [comment]
I gotta try this remote. Not sure how to do it yet but I will read up. However, I was messing around with my own remote and was curious if possible to have premade buttons for specific channels?  I only watch about 10 to 15 channels and could see having a list of those channels. Rather than his 5 7 0, his 570 and it sends all three numbers in a row. I could even name it ESPN, for example.

**Reply by GerryDazoo** [reply]
I'm probably not going to do it, but there's nothing stopping you from making your own webpage that will do a POST with the channel info in it. I haven't tried anything that fancy but it's theoretically possible with a custom remote definitions file. You can stick in whatever HTML you want between the button definitions.

**Reply by GerryDazoo** [reply]
Turns out you can't stick in a new input field due to an implemention issue. I'll make a small change to support this. Standby.

**Reply by GerryDazoo** [reply]
update committed. See the release notes. Here's an example
       'MSNBC' : 9,13,13

**Comment by DoogieHowitzerWMD** [comment]
I've found fios is mostly compatible with the default remote Gerry has, but I'll give this a try when I have the chance.

**Reply by socelillyhammer** [reply]
@DoogieHowitzerWMD  which fios box do you have - the new ones the ipc4100 and ipc1100?  I am having issues getting my remote to work

**Reply by DoogieHowitzerWMD** [reply]
I have a few of the Motorolla ones. QIP 7100, 7216, and 7232

Gerry's default remote file is like 90% compatible with it, so that's what I use. The other Fios remotes people have made have slightly better keymapping (for some of the more ancillary functions that I don't use), but the channel numbers were at the bottom, which caused the page to reload back to the top after a key press. Now that we can pass the channel number in the URL, that might not be the biggest issue anymore though.

I saw this after I replied in the other thread with the keycode database (so the ipc boxes likely aren't there). Do they have radically different remotes than the Motorola boxes?

**Reply by socelillyhammer** [reply]
The remotes are different for IPC4100 and 1100. Oddly last year SB manufacturer list was able to work but it had your remote the UI the p265.  But I think the mapping is different, othewise some buttons on the default remote would have worked.  I did use the txt file of the other remotes to bring up their UI but the mapping reads similar to the default remote - unless Im reading things wrong.

The RT button to change and remap the IR code in the default remote is no longer present (unless im doing something wrong) so Im going to try and go though #165 to see what I find, learn how to read code to the best of my ability and try and piece this all together.  What seems like 5 minutes for me is an 2 hours of real time when Im trying to learn the lanugage.

But I appreciate pointing me to that discussion. Its a small step forward

**Reply by socelillyhammer** [reply]
@DoogieHowitzerWMD from quickly scanning some vzw remotes and comparing the mapping to the default remote, the codes and numbers are differnt.  For example default has only "1" where vzw has "001"  Im changing the mapping to corrolate but im doing something wrong as i dont see the right number appear in teh server when I hit the button i need to correspond.

So far its a big start, but like i said, I need to learn to read code. Cant thank you enough for pointing me in the right direction.

**Comment by socelillyhammer** [comment]
Is there a way to check if my IR reciever is working?

I've compared the remote map logs #165  against the default remote in slinGER for verizon fios. The code numbers more or less match up and at the very least some of the buttons should work for me, but they dont. So I know thats not my issue.

I'm wondering if there is a way to test my IR receiver or if its bricked or the settings before SB went dark are stuck on a device that doesnt work with my STB. And if the last one is the case, can it be fixed? or would a hard reset work (now that I have my ID and password)
